geotorch/hurwitz.py

```diff
@@ -0,0 +1,164 @@
+import torch
+from .psd import PSD
+from .skew import Skew
+from .product import ProductManifold
+from .exceptions import VectorError, NonSquareError
+from .utils import _extra_repr
+from .exceptions import (
+    VectorError,
+    NonSquareError,
+    InManifoldError,
+)
+
+def get_lyap_exp(A):
+    return -torch.max(torch.real(torch.linalg.eigvals(A)))
+
+class Hurwitz(ProductManifold):
+
+    def __init__(self, size, alpha: float = 0.0, triv="expm"):
+        r"""
+        Manifold of matrices with eigenvalues with negative real parts,
+        also called Hurwitz matrices. They represend linear stable linear dynamical systems.
+
+        We have the following result : A is Hurwitz with prescribed decay rate \alpha
+        if and only if it can be written as
+        :math:`A = P^{-1}(-\frac{Q}{2} +S) - \alpha I_n`
+        with $P > 0, Q > 0$ and ^S^skew-symmetric
```

Comment:
What does `P > 0` mean here? Do you mean `P \in PSD(n)`? Please follow the notation in the rest of the library.

Reply:
Also, Fix the skew-symmetric notation.